sway/config/output.c

```diff
@@ -164,23 +164,31 @@ static void set_mode(struct wlr_output *output, int width, int height,
 	}
 }
 
-void terminate_swaybg(pid_t pid) {
-	int ret = kill(pid, SIGTERM);
-	if (ret != 0) {
-		wlr_log(WLR_ERROR, "Unable to terminate swaybg [pid: %d]", pid);
-	} else {
-		int status;
-		waitpid(pid, &status, 0);
+static void terminate_swaybg(struct wl_display *display, pid_t pid) {
+	struct wl_client *client;
+	wl_client_for_each(client, wl_display_get_client_list(display)) {
+		pid_t p;
+		wl_client_get_credentials(client, &p, NULL, NULL);
+
+		if (p == pid) {
+			wl_client_destroy(client);
+			break;
```

Comment:
`swaybg` can be set to a third-party client, e.g. oguri. This client can open multiple connections. We shouldn't break here.